test/test_nn.py

```diff
@@ -21263,6 +21265,21 @@ def hook_with_module(module, state_dict, prefix, local_metadata, strict, missing
         m_load.load_state_dict(m_state_dict)
         self.assertEqual(2, hook_called)
 
+    def test_no_extra_ref_to_module(self):
+        gc.disable()
+        m = nn.Linear(10, 10)
+
+        def hook_with_module(*args, **kwargs):
+            pass
+
+        m._register_load_state_dict_pre_hook(hook_with_module, True)
+        weak_m = weakref.ref(m)
+        del m
+
+        self.assertEqual(weak_m(), None)
+        gc.enable()
```

Comment:
nit:
```python
try:
  gc.disable()
  # rest of the test
finally:
 gc.enable()
```
We don't want a failure in this test to disable the gc for all other tests!

Reply:
That's a very good nit! I just pushed a change doing exactly this